src/schoology-extractor/tests/api/test_request_client.py

```diff
@@ -228,4 +233,34 @@ def test_given_correct_parameter_then_url_is_built_correctly(self, default_reque
             # Assert
             assert result == expected_result
 
+    class Test_when_get_grading_periods_method_is_called:
+        def test_given_wrong_type_in_params_then_thrown_assertion_error(self, default_request_client):
+            with pytest.raises(AssertionError):
+                default_request_client.get_grading_periods([])
+
+        def test_given_correct_parameter_then_get_call_is_made(self, default_request_client, mocker):
+            # Arrange
+            get_mock = mocker.MagicMock(return_value={})
+            default_request_client.get = get_mock
+
+            # Act
+            default_request_client.get_grading_periods()
+
+            # Assert
+            assert get_mock.call_count == 1
```

Comment:
Wouldn't it be more valuable to ensure that the expected results are returned, rather than only confirming that the method was called?

Reply:
You are right, I'll fix it

Reply:
✅

Reply:
Done ✅